File: network/src/main/java/pl/malopolska/smoksmog/model/Particulate.java
package pl.malopolska.smoksmog.model;

import com.google.gson.annotations.SerializedName;

import org.joda.time.DateTime;

import java.util.ArrayList;
import java.util.List;

import lombok.Getter;

@Getter
public class Particulate {

    private long id;
    private String name;
    @SerializedName( "short_name" )
    private String shortName;
    private float value;
    private String unit;
    private float norm;
    private DateTime date;
    @SerializedName( "avg" )
    private float average;
    private int position;

    private List<History> values = new ArrayList<>();
}

File: network/src/main/kotlin/pl/malopolska/smoksmog/model/Particulate.kt
package pl.malopolska.smoksmog.model

import com.google.gson.annotations.SerializedName
import org.joda.time.DateTime
import java.util.*

data class Particulate(
        val id: Long,
        val name: String,
        @SerializedName("short_name") val shortName: String,
        val value: Float = 0f,
        val unit: String,
        val norm: Float = 0f,
        val date: DateTime,
        @SerializedName("avg") val average: Float = 0f,
        val position: Int = 0,
        val values: List<History> = ArrayList<History>()) {

    val enum: ParticulateEnum
        get() = ParticulateEnum.findById(id)
}
